Add JPEG support

eventualbuddha opened the pull request:
Closes #2

kAworu commented:
Hello @eventualbuddha,

Thanks a lot for this PR. I'm taking a look at it and overall it look good to me.

As a library, I think we should avoid printing on stderr (on jpeg error). Once that is gone I'll be happy to merge and make a new release!

eventualbuddha commented:
@kAworu I removed the stderr print. I had pretty much just copied all the code from quirc so I didn't really think too much about the printing, but you're right. I also updated the README to note that JPEG is now supported.

eventualbuddha commented on src/node_quirc_decode.c:
```diff
@@ -311,3 +316,83 @@ nq_load_png(struct quirc *q, const uint8_t *img, size_t img_len)
+	char buf[JMSG_LENGTH_MAX];
+
+	err->base.format_message(com, buf);
+	fprintf(stderr, "JPEG error: %s\n", buf);
```
Whoops, I missed this one.

eventualbuddha replied:
Okay, fixed.
